### DISCUSSION TITLE
TLS_Level

### DISCUSSION THREAD

**Discussion by jarekglo78** [docker-mailserver/docker-mailserver#2984] [Q&A] [answered]
Hello

I use mailserver/docker-mailserver at TrueNas.
It work correct. Thank you.

I checked my mail server at ssllabs.com and my server use/support TLSv1.2 and TLSv1.1 and TLSv1.

I set TLS_Level=modern in Environment Variables and tested again. SSLlabs test still report TLSv1 and TLSv1.1 are support.

How do I can turn off TLSv1 and TLSv1.1 support in my server.

Regards
Jarek

**Comment by casperklein** [comment, marked as answer]
ssllabs.com only tests HTTPS, not SMTP 😉 

You may take a look at the email test on https://internet.nl/

**Comment by jarekglo78** [comment]
You are right.
Thanks for your fast answer. 

I tested, I must work more.

**Comment by polarathene** [comment]
You can also use [`testssl.sh`](https://github.com/drwetter/testssl.sh) if you're wanting a similar test that can test TLS against SMTP ports.

Remember that these are tools to provide insights, not chase a grade without understanding why it is not the top grade.

---

For reference, we run the following tests ourselves in CI:
https://github.com/docker-mailserver/docker-mailserver/blob/master/test/test-files/ssl/example.test/testssl.txt

That config file is provided to `testssl.sh` docker image here:
https://github.com/docker-mailserver/docker-mailserver/blob/88715974eb45bdf1fb189292685f408dac237d96/test/tests/parallel/set2/tls/tls_cipherlists.bats#L143-L172

We presently only use it to ensure our TLS cipher suites (_only relevant to TLS 1.2 with DMS v12 onwards_) are what they should be (_differs not only by TLS_LEVEL and cert type, but presently port 25 has a different set of cipher suites permitted_).

In the past, I have audited the TLS support where I went over `testssl.sh` results and various other tools and guides / advice with best practices to ensure DMS was configured properly for TLS. If you find this is no longer the case, or have any queries, do let us know :+1:
